pkg/web_app/lib/src/account.dart

```diff
@@ -18,31 +19,53 @@ import 'google_js.dart';
 import 'page_data.dart';
 import 'pubapi.client.dart';
 
-bool _initialized = false;
+final _initialized = Completer<void>();
 GoogleUser _currentUser;
 
 /// Returns whether the user is currently signed-in.
-bool get isSignedIn => _initialized && _currentUser != null;
+bool get isSignedIn => _initialized.isCompleted && _currentUser != null;
 
 /// Returns the currently signed-in user or null.
 GoogleUser get currentUser => _currentUser;
 
-PubApiClient _client;
-http.Client _httpClient;
 final _pkgAdminWidget = _PkgAdminWidget();
 final _publisherAdminWidget = _PublisherAdminWidget();
 final _createPublisherWidget = _CreatePublisherWidget();
 final _consentWidget = _ConsentWidget();
 
+String get _baseUrl {
+  final location = Uri.parse(window.location.href);
+  return Uri(
+    scheme: location.scheme,
+    host: location.host,
+    port: location.port,
+  ).toString();
+}
+
+/// The pub API client to use without credentials.
+PubApiClient get unauthenticatedClient =>
+    PubApiClient(_baseUrl, client: http.BrowserClient());
+
 /// The pub API client to use with account credentials.
 PubApiClient get client {
-  if (_client == null) {
-    _httpClient ??=
-        getAuthenticatedClient(currentUser?.getAuthResponse(true)?.id_token);
-    final uri = Uri.parse(window.location.href);
-    _client = PubApiClient(uri.resolve('/').toString(), client: _httpClient);
-  }
-  return _client;
+  return PubApiClient(_baseUrl, client: createAuthenticatedClient(() async {
```

Comment:
Shouldn't we cache the instance? Or the BrowserClient is a throw-away instance anyway, and we don't gain anything from it?

Reply:
I haven't looked at `BrowserClient`, but if it's implemented in a browser I doubt it can cache anything... As connections are managed by the browser.

If any event, if we profile performance issues then I suggest we cache, until then head -> sand :see_no_evil: 

(seriously, though, I/O is the expensive part here, I doubt the rest has much cost).